kobo/apps/trackers/tests.py

```diff
@@ -0,0 +1,94 @@
+import datetime
+
+from django.contrib.auth.models import User
+
+from kobo.apps.trackers.models import MonthlyNLPUsageCounter
+from kobo.apps.trackers.utils import update_nlp_counter
+from kpi.models.asset import Asset
+from kpi.tests.kpi_test_case import KpiTestCase
+
+
+class TrackersTestCases(KpiTestCase):
+    fixtures = ['test_data']
+
+    def setUp(self):
+        self.client.login(username='someuser', password='someuser')
+        self.user = User.objects.get(username='someuser')
+        self.TEST_TRANSCRIPTION_SERVICES = [
+            'acme_1_speech2text',
+            'optimus_transcribers',
+            'wonka_stenographers',
+        ]
+        self.today = datetime.datetime.today()
+
+    def _create_asset(self):
+        asset = Asset.objects.create(
+            content={'survey': [{"type": "text", "name": "q1"}]},
+            owner=self.user,
+            asset_type='survey',
+            name='тєѕт αѕѕєт'
+        )
+        asset.deploy(backend='mock', active=True)
+        asset.save()
+        return asset
+
+    def test_asset_deletion(self):
+        asset = self._create_asset()
+        asset.deploy(backend='mock', active=True)
+        asset.save()
+        MonthlyNLPUsageCounter.objects.create(
+            year=self.today.year,
+            month=self.today.month,
+            user=self.user,
+            asset_id=asset.id,
+            counters={'some_key': 4504},
+        )
+        assert MonthlyNLPUsageCounter.objects.all().count() == 1
+        a_id = asset.id
+        asset.delete()
+        assert MonthlyNLPUsageCounter.objects.all().count() == 1
+        tracker_no_asset = MonthlyNLPUsageCounter.objects.filter(
+            user=self.user,
+            asset_id=None,
+        )
+        tracker_with_asset = MonthlyNLPUsageCounter.objects.filter(
+            asset_id=a_id,
+        )
+        assert tracker_no_asset.count() == 1
```

Comment:
Maybe we should assert it equals 0 before calling `asset.delete()` . 
`assert MonthlyNLPUsageCounter.objects.all().count() == 1` line 49 does not make it obvious that the existing one if for `asset = None`.